main/webapp/modules/core/scripts/views/data-table/data-table-view.js

```diff
@@ -493,20 +513,85 @@ DataTableView.prototype._showRows = function(start, onDone) {
   }, this._sorting);
 };
 
+DataTableView.prototype._onChangeGotoPage = function(elmt, evt) {
+  var gotoPageNumber = parseInt($('input#viewpanel-paging-current-input').val());
+  
+  if(typeof gotoPageNumber != "number" || isNaN(gotoPageNumber) || gotoPageNumber == "") { 
+    $('input#viewpanel-paging-current-input').val(this._currentPageNumber); 
+    return;
+  }
+  
+  if(gotoPageNumber > this._lastPageNumber) gotoPageNumber = this._lastPageNumber;
+  if(gotoPageNumber < 1) gotoPageNumber = 1;
+  
+  this._currentPageNumber = gotoPageNumber;
+  this._showRows((gotoPageNumber - 1) * this._pageSize);
+};
+
+DataTableView.prototype._onKeyDownGotoPage = function(elmt, evt) {
+  var keyDownCode = event.which;
+  
+  if([38, 40].indexOf(keyDownCode) == -1) return;
+  
+  evt.preventDefault();
+
+  var newPageValue = $('input#viewpanel-paging-current-input')[0].value;
+  if(keyDownCode == 38) {  // Up arrow
+    if(newPageValue <= 1) return;
+    this._onClickPreviousPage(elmt, evt);
+  }
+    
+  if(keyDownCode == 40) {  // Down arrow
+    if(newPageValue >= this._lastPageNumber) return;
+    this._onClickNextPage(elmt, evt);
+  }
+  
+  // For the arrow keys, in order to have a keyDownRepeat, we have to put the focus back on the « currentPageInput »
+  // The <input> seams to be deleted and recreated, so we wait with a few setTimeout() until it is there, unless 1 s. elapse, then we stop
+  const refocusDelay = 50;
```

Comment:
By principle I do not think we should start relying on timeouts like this to build the UI. If the DOM element is deleted and re-added somewhere, then the code to bind events on it should be placed there, in the code that sets it up.

Reply:
> By principle I do not think we should start relying on timeouts like this to build the UI.
Yes, I agree with you that.

> If the DOM element is deleted and re-added somewhere, then the code to bind events on it should be placed there, in the code that sets it up.

@wetneb: rebinding is not the problem, it's probably already done. What we have here is the need to « refocus » inside the next input field.

Me, I think the best way would just be to not delete the input, then, you have nothing to do more. The setTimeout is just compensation required by that fact that the paging control are deleted/recreated each time a page changes. It could be much more gentle in the code.

So, I guess we should just drop this « Infinite Paging » with the arrow keys. It's not important, and I coded it because @thadguidry wanted to try that, but it is beyond the scope of this issue. When @lisa761 will be done with the table code overhaul, I can revisit this. But now, either we live with it as is, or we remove it. I doesn't break at all on Chrome, so it's not a bad coding.

Maybe I don't understand what you mean by « then the code to bind events on it should be placed there, in the code that sets it up. » because from what I see, it gets rebinded after being rebuilt.

Regards,
   Antoine

Reply:
What I mean is that we should not rely on any timeouts to refocus / rebind / do any other type of set-up on the UI. Just use callbacks that are executed when some elements are created / deleted / other lifecycle events. This is just to avoid building up technical dept, running into situations were we have to deal with various race conditions between the timeouts, and so on.

Reply:
> What I mean is that we should not rely on any timeouts to refocus / rebind / do any other type of set-up on the UI. Just use callbacks that are executed when some elements are created / deleted / other lifecycle events. This is just to avoid building up technical dept, running into situations were we have to deal with various race conditions between the timeouts, and so on.

@wetneb: ok, I get something in my mind now with what you write. K. A very simple fix would be: everytime the page is loaded, that focus goes to the field. That way, I could just remove my coding, and all will work.

What do you think of that?

Regards,
   Antoine

Reply:
It does sound better to me!

Reply:
> It does sound better to me!

Et voilà !

@wetneb: It's fixed. And now, it even works perfectly in FF, not just Chrome.

Now, I guess it's done! ;-)

Regards,
   Antoine

Reply:
Yeap, can confirm this works well in FF now for me.